src/SerialFirmata.cpp

```diff
@@ -296,27 +310,43 @@ void SerialFirmata::checkSerial()
         continue;
       }
 #endif
-      if (serialPort->available() > 0) {
-        Firmata.write(START_SYSEX);
-        Firmata.write(SERIAL_MESSAGE);
-        Firmata.write(SERIAL_REPLY | portId);
-
-        if (bytesToRead == 0 || (serialPort->available() <= bytesToRead)) {
-          numBytesToRead = serialPort->available();
-        } else {
-          numBytesToRead = bytesToRead;
+      int availableBytes = serialPort->available();
+      if (availableBytes > 0) {
+        bool read = true;
+        
+#if defined(SERIAL_STORE_AND_FORWARD)    
+        // check if reading should be delayed to collect some bytes before forwarding (for baud rates up to 19200)
+        if (maxCharDelay[portId]) {          
+          // inter character delay exceeded or more than 48 bytes available or more bytes available than required          
```

Comment:
I expect hardware serial and software serial ports to have 64 bytes of receive buffer. Processing a Firmata loop requires a variable amount of time. I chose this value to stay clear of a buffer overrun but this is not completely foolproof.